### ISSUE TITLE
Bug: The token type of a private name should be PrivateIdentifier instead of Identifier and the value should be the string value of a private name

### ISSUE THREAD

**Issue by JLHwung** [typescript-eslint/typescript-eslint#11022]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

ast-spec

### Playground Link

https://typescript-eslint.io/play/#ts=5.6.2&showAST=es&fileType=.tsx&code=MYGwhgzhAEDC0G9oGIAOBuaBfIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=true

### Repro Code

```TypeScript
class C { #p; }
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/<rule-name>": ["error", ...<options>],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

`.ast.tokens[3].type` should be `PrivateIdentifier` and `.ast.tokens[3].value` should be `"p"`.

### Actual Result

`.ast.tokens[3].type` is `Identifier` and `.ast.tokens[3].value` is `#p`.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/typescript-estree` | 8.28.0 |
